web/static/js/components/questionnaires/QuotaCompletedMsg.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component, PropTypes } from 'react'
+import { connect } from 'react-redux'
+import { InputWithLabel, Card } from '../ui'
+import * as actions from '../../actions/questionnaire'
+
+class QuotaCompletedMsg extends Component {
+  static propTypes = {
+    dispatch: PropTypes.func.isRequired,
+    questionnaire: PropTypes.object
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      editing: false
+    }
+    this.quotaCompletedMsgChanged = this.quotaCompletedMsgChanged.bind(this)
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick() {
+    this.setState({editing: !this.state.editing})
+  }
+
+  quotaCompletedMsgChanged(e, mode) {
+    const { dispatch } = this.props
+    e.preventDefault()
+    switch (mode) {
+      case 'sms':
+        dispatch(actions.setSmsQuotaCompletedMsg(e.target.value))
+        break
+      case 'ivr':
+        dispatch(actions.setIvrQuotaCompletedMsg(e.target.value))
+        break
+    }
+  }
+
+  render() {
+    const { questionnaire } = this.props
+
+    let quotaCompletedMsgs = questionnaire.modes.map((mode) => {
+      const value = questionnaire.quotaCompletedMsg && questionnaire.quotaCompletedMsg[questionnaire.defaultLanguage][mode] ? questionnaire.quotaCompletedMsg[questionnaire.defaultLanguage][mode] : ''
```

Comment:
This feature is to be supported just for the questionnaire's default language for now?